### ISSUE TITLE
Support for Semi-Structured XML

### ISSUE THREAD

**Issue by margru** [martinblech/xmltodict#63]
Semi-structured element should not be parsed and the content should be kept as is. 
For example:

``` XML
<level1>
    <level2>
        <level3>First level3.</level3>
        text outside 1st level3 at the end
    </level2>
    <level2>
        text outside 2nd level3 at the beginning
        <level3>Second level3.</level3>
        text outside 2nd level3 at the end
    </level2>
    <level2>
        text outside 3rd level3 at the beginning
        <level3>Third leve3.</level3>
    </level2>
</level1>
```

will produce (at the corresponding levels):

``` JSON
level2: [
{ #text: "text outside 1st level3 at the end", level3: "First level3." },
{ #text: "text outside 2nd level3 at the beginningtext outside 2st level3 at the end", level3: "Second level3." },
{ #text: "text outside 3rd level3 at the end", level3: "Third level3." },
]
```

which is not only irreversible (not keeping order) but for _2nd level3_ also meaningless (joining the texts). According to ["spec"](http://www.xml.com/pub/a/2006/05/31/converting-between-xml-and-json.html?page=2) you claim you are adopting, at least the case of _2nd level3_ should not be parsed.

It would be also great if a tag name(s) could be specified (as a parameter to parse function) whose content wouldn't be parsed at all. It could also solve the described issue sometimes (as the user would specify that tag `level2` shouldn't be parsed and its content should be kept in `#text` property).

Thank you in advance for your comments.

**Comment by beyondai** [comment]
Hi all,

You can hack through this issue by replacing the tag in the response string before passing to xmltodict parser.
For example, if your text is "<someTag>bar foo</someTag>" and you do not want it to be parsed, you can try xmltodict.parse(resp.text.replace("<someTag>", "").replace("</someTag>", "")).
This simple hack does not solve all problems, but it solves mine.

Thanks,

**Comment by raoulschaffner** [comment]
i'm also very interested in a fix to this problem. i have to deal with semi-structured bibliographical data in xml that contains elements with html-formatted content.

**Comment by javadev** [comment]
It may be thus output for xml:

```javascript
{
  "level1": {
    "level2": [
      {
        "level3": "First level3.",
        "#text": "\n        text outside 1st level3 at the end\n    "
      },
      {
        "#text": "\n        text outside 2nd level3 at the beginning\n        ",
        "level3": "Second level3.",
        "#text1": "\n        text outside 2nd level3 at the end\n    "
      },
      {
        "#text": "\n        text outside 3rd level3 at the beginning\n        ",
        "level3": "Third leve3."
      }
    ]
  }
}
```

**Comment by margru** [comment]
@javadev This is not true. AFAIK, JSON specification doesn't define the order so that you cannot be sure what is the correct order of keys "#text", "level3" and "#text1" (it means that in your proposal the XML output can have different order than the original XML). You need to store the items as lists somehow to keep the order.

**Comment by javadev** [comment]
It depends on json parser implementation.
If json parser stores element's order it will work.

**Comment by margru** [comment]
@javadev Yes but it depends on the implementation. We are talking about JSON specification.

**Comment by javadev** [comment]
I agree. In case of external independent json parser we get different elements order.
